login-utils/login.c

```diff
@@ -1520,6 +1520,7 @@ int main(int argc, char **argv)
 	}
 
 	child_argv[child_argc++] = NULL;
+	free_getlogindefs_data();
 
```

Comment:
I have doubts this is a good idea. After fork() Linux kernel creates a new process with copy-on-write heap pages, I guess your free() call(s) will modify this area and it will trigger pages coping. It seems better to rely on fork()+exec() Linux efficiency. We nowhere cleanup (except file descriptors, etc) anything between fork() and exec().

Reply:
And it's also possible that malloc() is implemented by mmap() anonymous pages and it does not preserve over an exec() ...

Reply:
I dropped this change from the patch set.